includes/Jobs/DataDumpGenerateJob.php

```diff
@@ -5,176 +5,360 @@
-use MediaWiki\MediaWikiServices;
 use MediaWiki\Shell\Shell;
-use MediaWiki\Title\Title;
+use MediaWiki\SpecialPage\SpecialPage;
 use MediaWiki\User\User;
-use MediaWiki\User\UserIdentity;
+use Miraheze\DataDump\ConfigNames;
 use Miraheze\DataDump\DataDump;
+use MWExceptionHandler;
+use RuntimeException;
+use Wikimedia\Rdbms\IConnectionProvider;
+use Wikimedia\Rdbms\IDatabase;
 
 class DataDumpGenerateJob extends Job {
 
-	/** @var Config */
-	private $config;
+	public const JOB_NAME = 'DataDumpGenerateJob';
 
-	public function __construct( $title, $params ) {
-		parent::__construct( 'DataDumpGenerateJob', $title, $params );
+	private Config $config;
+	private IConnectionProvider $connectionProvider;
 
-		$this->config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'DataDump' );
-	}
+	private array $arguments;
+	private string $fileName;
+	private string $type;
 
-	private function log( UserIdentity $user, string $action, string $fileName, ?string $comment = null ) {
-		$logEntry = new ManualLogEntry( 'datadump', $action );
-		$logEntry->setPerformer( $user );
-		$logEntry->setTarget( Title::newFromText( 'Special:DataDump' ) );
+	public function __construct(
+		array $params,
+		ConfigFactory $configFactory,
+		IConnectionProvider $connectionProvider
+	) {
+		parent::__construct( self::JOB_NAME, $params );
 
-		if ( $comment ) {
-			$logEntry->setComment( $comment );
-		}
+		$this->arguments = $params['arguments'];
+		$this->fileName = $params['fileName'];
+		$this->type = $params['type'];
 
-		$logEntry->setParameters( [ '4::filename' => $fileName ] );
-		$logEntry->publish( $logEntry->insert() );
+		$this->config = $configFactory->makeConfig( 'DataDump' );
+		$this->connectionProvider = $connectionProvider;
 	}
 
-	public function run() {
-		$dataDumpConfig = $this->config->get( 'DataDump' );
-		$dataDumpLimits = $this->config->get( 'DataDumpLimits' );
+	public function run(): bool {
+		$dataDumpConfig = $this->config->get( ConfigNames::DataDump );
+		$dataDumpLimits = $this->config->get( ConfigNames::Limits );
 		$dbName = $this->config->get( MainConfigNames::DBname );
+		$dbw = $this->connectionProvider->getPrimaryDatabase();
 
-		$dbw = MediaWikiServices::getInstance()
-			->getDBLoadBalancer()
-			->getMaintenanceConnectionRef( DB_PRIMARY );
+		$fileName = $this->fileName;
+		$type = $this->type;
 
-		$fileName = $this->params['fileName'];
-		$type = $this->params['type'];
+		$this->setStatus(
+			status: 'in-progress',
+			dbw: $dbw,
+			directoryBackend: '',
+			fileName: $fileName,
+			fname: __METHOD__,
+			comment: '',
+			fileSize: 0
+		);
 
-		$this->setStatus( 'in-progress', $dbw, '', $fileName, __METHOD__ );
-
-		$options = [];
-		foreach ( $dataDumpConfig[$type]['generate']['options'] as $option ) {
-			$options[] = preg_replace( '/\$\{filename\}/im', $fileName, $option );
-		}
+		$options = array_map(
+			fn ( string $opt ): string => preg_replace(
+				'/\$\{filename\}/im', $fileName, $opt
+			),
```

Comment:
I don't think we need mode `m`, so try [`str_ireplace`](https://www.php.net/manual/en/function.str-ireplace.php) (or perhaps even `str_replace`, but that might break API)

Reply:
I think I will just leave it for now as this currently just maintains how it is done currently.